### PR TITLE
Clarify and fix account related endpoints

### PR DESCRIPTION
**Pull request by fractallian**
When Nylas.config is called, create a Nylas.accounts endpoint that hits /a/<appid>/accounts and uses a special connection that uses app secret instead of access token. This fetches info about your app's accounts. This endpoint is undocumented?

This also fixes the connection.account endpoint to behave like all others. It fetches info about the connection's account. https://nylas.com/docs/platform/#account

This fixes issue #1

### CODE REVIEW

File: spec/delta-spec.coffee [hunk 1]
```diff
@@ -3,6 +3,7 @@ jasmine = require 'jasmine-node'
 {PassThrough} = require 'stream'
 Delta = require '../models/delta'
 NylasConnection = require '../nylas-connection'
+Promise = require 'bluebird'
```
Comment (khamidou) [inline spec/delta-spec.coffee]: Why add this line?

**Comment by khamidou** [comment]
Thanks for fixing this, @fractallian! It makes the code much cleaner.
I only have two remarks I'd like you to address before I merge this pull request:

1. Could you sign our [Contributor License Agreement](https://www.nylas.com/cla.html). It's similar to the ones used by Meteor and NodeJS.
2. I'd prefer that `ManagementAccounts` belonged to a `ManagementModelCollection`, instead of making `RestfulModelCollection` handle both accounts and the other models. Could you change this?

regards,

Karim

File: spec/delta-spec.coffee (hunk 1 above)
Comment (fractallian) [inline spec/delta-spec.coffee]: I was getting a Promise undefined error when running the test.

**Comment by fractallian** [comment]
You're welcome! Happy to sign the CLA and make your desired change.

**Comment by fractallian** [comment]
On second look, I had intended to remove `ManagementModelCollection` entirely. It is no longer needed after moving `ManagementAccounts` to top level `Nylas` and using a separate connection.

**Comment by TimFletcher** [comment]
I'm struggling to get the account info for a connected account and I believe this fixes it. Could this be merged please? Thanks!

**Comment by jstejada** [comment]
Hey @fractallian, @qbalin, @mccolin, thanks for filing and updating this PR, and sorry for the delay in getting back to you!
It seems that this PR is from a branch that you are separately maintaining and contains commits from PR #6 which has been closed and won't be included (added #40 instead), and other commits for unrelated changes.

It seems that some of these changes are already on master, but it would still be great if we could split the changes here into separate PRs, and I can go ahead and merge them in individually. Otherwise I can cherry pick any of the changes that are still relevant and file them as PRs independently. 

Going to close this one for now. Thanks!

**Pull request closed without merging**